code/crates/test/tests/it/wal.rs

```diff
@@ -289,7 +291,7 @@ async fn byzantine_proposer_crashes_after_proposing_1(params: TestParams) {
             TestParams {
                 enable_value_sync: true,
                 timeout_step: Duration::from_secs(5),
-                value_payload: ValuePayload::ProposalAndParts,
+                value_payload: ValuePayload::PartsOnly,
```

Comment:
Not sure why we need this change. Looking at the test app I think in general it's more accurate to use `ProposalAndParts` because for re-streaming the `Init` doesn't seem to carry `valid_round` so the receiver cannot reconstruct the `valid_round` without an explicit Proposal message.
For the purpose of these tests it's fine but I think for other cases we should be able to test with both.

Reply:
I had to do this because of this check in the test app:

https://github.com/informalsystems/malachite/blob/7e6c49dc8f39dcc840688da7fbbcae44c14fe5b4/code/crates/test/app/src/app.rs#L127-L131

Reply:
But this looks arbitrary, let me attempt to remove it and re-enable to proposal-and-parts tests.